lib/sublime-block-comment.js

```diff
@@ -11,15 +11,31 @@ export default {
     const buffer = editor.getBuffer();
 
     buffer.transact(() => {
-      for (const selection of editor.getSelectionsOrderedByBufferPosition().reverse()) {
-        const range = selection.getBufferRange();
-        const textInRange = buffer.getTextInRange(range);
-
+      for (let i in editor.getSelectionsOrderedByBufferPosition().reverse()) {
+        const selection = editor.getSelectionsOrderedByBufferPosition().reverse()[i];
+        const langs = editor.scopeDescriptorForBufferPosition(selection.getBufferRange().start).getScopesArray();
         // TODO change comment syntax based on scope's grammar
-        const commentTokens = {
+        let commentTokens = {
           open:  '/*',
-          close: '*/',
+          close: '*/'
         };
+        for (let lang of langs) {
+          if (lang.indexOf('text.html') > -1) {
+            commentTokens = {
+              open:  '<!--',
+              close: '-->',
+            };
+          }
+          if ((lang.indexOf('source.js.embedded.html') > -1) ||
```

Comment:
Yes, that can work.
I'm thinking whether we can use a more generic approach, like reverse searching (from the end to beginning of the scopes array) for a scope that begins with `source.jss`/`source.css`.
I'll try to implement it and see if it works.
Perhaps I'm overcomplicating things.